src/components/PokemonScreen/PokemonScreen.tsx

```diff
@@ -15,24 +15,65 @@ import {
   ChevronRightIcon,
 } from "@chakra-ui/icons";
 import { usePokemonContext } from "../../hooks/usePokemonContext";
+import { useEffect, useState } from "react";
 
 interface Props {
   isOpen: boolean;
   onClose: () => void;
 }
 
 function PokemonScreen({ isOpen, onClose }: Props) {
-  const { pokemon } = usePokemonContext();
-  const firstType = pokemon.types[0].type.name;
+  const { pokemon, setPokemon, pokemonList } = usePokemonContext();
+  const [slideAnimation, setSlideAnimation] = useState("");
+  const firstType = pokemon.types[0];
+
+  function handleNextPokemon() {
+    const nextPokemon = pokemonList.find((p) => p.id === pokemon.id + 1);
+    if (nextPokemon) {
+      setSlideAnimation("slide-left");
+      setTimeout(() => {
+        setPokemon(nextPokemon);
+        setSlideAnimation("");
+      }, 100); 
+    }
+  }
+
+  function handlePreviousPokemon() {
+    const previousPokemon = pokemonList.find((p) => p.id === pokemon.id - 1);
+    if (previousPokemon) {
+      setSlideAnimation("slide-right");
+      setTimeout(() => {
+        setPokemon(previousPokemon);
+        setSlideAnimation("");
+      }, 100); 
+    }
+  }
+
+  useEffect(() => {
+    const handleKeyDown = (event: any) => {
+      if (event.key === "ArrowRight") {
+        handleNextPokemon();
+      } else if (event.key === "ArrowLeft") {
+        handlePreviousPokemon();
+      }
+    };
+
+    window.addEventListener("keydown", handleKeyDown);
+
+    return () => {
+      window.removeEventListener("keydown", handleKeyDown);
+    };
+  }, [handleNextPokemon, handlePreviousPokemon]);
```

Comment:
Why are these methods being used as dependencies for the useEffect?

Reply:
I've fixed this deleting all the useEffect and handleKeyDown function. The onKeyDown prop in modalContent is being used to make all the keydown event job.